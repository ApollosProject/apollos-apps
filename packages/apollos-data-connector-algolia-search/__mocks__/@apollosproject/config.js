const ApolloServer = require.requireActual('@apollosproject/config').default;

ApolloServer.loadJs({
  ROCK: {
    TIMEZONE: 'America/New_York',
  },
  ALGOLIA: {
    APPLICATION_ID: 'some-application-id',
    API_KEY: 'some-api-key',
    CONFIGURATION: {},
  },
});

export default ApolloServer;
