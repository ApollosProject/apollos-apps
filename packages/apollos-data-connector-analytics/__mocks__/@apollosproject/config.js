const ApollosConfig = require.requireActual('@apollosproject/config');

const defaults = {
  ANALYTICS: { SEGMENT_KEY: 'something', GA_ID: 'something-else' },
};

ApollosConfig.default.loadJs(defaults);

console.log(ApollosConfig);

class dataSource {
  constructor() {
    this.config = null;
    this.loadJs = this.loadJs.bind(this);
  }

  initialize({ context }) {
    if (context?.church?.slug) {
      const config = new ApollosConfig.Config();
      config.loadJs(defaults);
      this.config = config;
      this.attachConfigToClass(this.config);
    }
  }

  loadJs(input) {
    this.config.loadJs(input);
    this.attachConfigToClass(this.config);
  }

  attachConfigToClass(config) {
    Object.keys({ ...config }).forEach((key) => {
      if (key !== 'config') {
        Object.defineProperty(this, key, {
          enumerable: true,
          configurable: true,
          writable: true,
          value: config[key],
        });
      }
    });
  }
}

export { dataSource };
export default ApollosConfig;
