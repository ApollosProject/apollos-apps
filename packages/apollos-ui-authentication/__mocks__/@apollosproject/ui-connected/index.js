export const RockAuthedWebBrowser = ({ children }) => children();
