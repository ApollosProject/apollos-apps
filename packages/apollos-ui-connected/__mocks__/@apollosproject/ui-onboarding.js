export const checkOnboardingStatusAndNavigate = () => null;
export const onboardingComplete = () => null;
