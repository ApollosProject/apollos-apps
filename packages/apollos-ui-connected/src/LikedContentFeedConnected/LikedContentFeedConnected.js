import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Query } from '@apollo/client/react/components';
import { get } from 'lodash';

import { View } from 'react-native';
import {
  UIText,
  BackgroundView,
  FeedView,
  Icon,
  styled,
  withTheme,
} from '@apollosproject/ui-kit';

import fetchMoreResolver from '../utils/fetchMoreResolver';
import ContentCardConnected from '../ContentCardConnected';

import GET_LIKED_CONTENT from './getLikedContent';

const ListEmptyContainer = styled(() => ({
  display: 'flex',
  flex: 1,
  height: 500,
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
}))(View);

const ListEmptyText = withTheme(({ theme }) => ({
  bold: true,
  style: {
    color: theme.colors.text.tertiary,
    paddingTop: theme.sizing.baseUnit * 0.5,
  },
}))(UIText);

const ListEmptyIcon = withTheme(({ theme }) => ({
  size: 42,
  name: 'heart',
  fill: theme.colors.text.tertiary,
}))(Icon);

const ListEmptyComponent = () => (
  <ListEmptyContainer>
    <ListEmptyIcon />
    <ListEmptyText>"Like" an item to see it here</ListEmptyText>
  </ListEmptyContainer>
);

/** A FeedView wrapped in a query to pull content data. */
class LikedContentFeedConnected extends PureComponent {
  static propTypes = {
    Component: PropTypes.oneOfType([
      PropTypes.node,
      PropTypes.func,
      PropTypes.object, // type check for React fragments
    ]),
    ContentCardComponent: PropTypes.oneOfType([
      PropTypes.node,
      PropTypes.func,
      PropTypes.object, // type check for React fragments
    ]),
    /** Functions passed down from React Navigation to use in navigating to/from
     * items in the feed.
     */
    navigation: PropTypes.shape({
      navigate: PropTypes.func,
    }),
  };

  static defaultProps = {
    Component: FeedView,
    ContentCardComponent: ContentCardConnected,
  };

  /** Function that is called when a card in the feed is pressed.
   * Takes the user to the ContentSingle
   */
  handleOnPress = (item) =>
    this.props.navigation.navigate('ContentSingle', {
      itemId: item.id,
      sharing: item.sharing,
    });

  getContent = (data) =>
    get(data, 'likedContent.edges', []).map((edge) => edge.node);

  fetchMore = (data, fetchMore, variables) =>
    fetchMoreResolver({
      collectionName: 'likedContent',
      fetchMore,
      variables,
      data,
    });

  render() {
    const { Component, ContentCardComponent } = this.props;

    return (
      <BackgroundView avoidHeader>
        <Query
          query={GET_LIKED_CONTENT}
          fetchPolicy="cache-and-network"
          variables={{ first: 20 }}
        >
          {({ loading, error, data, refetch, fetchMore, variables }) => (
            <Component
              ListItemComponent={ContentCardComponent}
              ListEmptyComponent={<ListEmptyComponent />}
              content={this.getContent(data)}
              isLoading={loading}
              error={error}
              refetch={refetch}
              onPressItem={this.handleOnPress}
              fetchMore={this.fetchMore(data, fetchMore, variables)}
            />
          )}
        </Query>
      </BackgroundView>
    );
  }
}

export default LikedContentFeedConnected;
