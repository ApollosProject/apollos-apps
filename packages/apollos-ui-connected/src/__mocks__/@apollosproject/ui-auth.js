export const ProtectedTouchable = ({ children }) => children;
