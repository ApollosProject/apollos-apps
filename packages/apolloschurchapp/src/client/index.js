import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { ApolloProvider } from 'react-apollo';
import { ApolloClient } from 'apollo-client';
import { ApolloLink } from 'apollo-link';
import SplashScreen from 'react-native-splash-screen';
import gql from 'graphql-tag';

import httpLink from './httpLink';
import clientStateLink from './clientStateLink';
import authLink from './authLink'; // eslint-disable-line
import cache, { ensureCacheHydration } from './cache';

const link = clientStateLink.concat(ApolloLink.from([authLink, httpLink]));

export const client = new ApolloClient({
  link,
  cache,
  queryDeduplication: true,
  shouldBatch: true,
});

client.onResetStore(clientStateLink.writeDefaults);

export const CACHE_LOADED = gql`
  query {
    cacheLoaded @client
  }
`;

export const MARK_CACHE_LOADED = gql`
  mutation markCacheLoaded {
    cacheMarkLoaded @client
  }
`;

class ClientProvider extends PureComponent {
  static propTypes = {
    client: PropTypes.shape({
      cache: PropTypes.shape({}),
    }),
  };

  static defaultProps = {
    client,
  };

  async componentDidMount() {
    try {
      await ensureCacheHydration;
    } catch (e) {
      throw e;
    } finally {
      if (SplashScreen && SplashScreen.hide) SplashScreen.hide();
      client.mutate({ mutation: MARK_CACHE_LOADED });
    }
  }

  render() {
    return <ApolloProvider {...this.props} client={client} />;
  }
}

export default ClientProvider;
