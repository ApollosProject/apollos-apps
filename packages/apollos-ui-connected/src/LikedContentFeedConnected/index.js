import LikedContentFeedConnected from './LikedContentFeedConnected';

export GET_LIKED_CONTENT from './getLikedContent';

export default LikedContentFeedConnected;
