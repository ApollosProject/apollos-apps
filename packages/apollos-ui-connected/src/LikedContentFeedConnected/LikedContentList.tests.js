import React from 'react';

import { GET_CONTENT_CARD } from '@apollosproject/ui-connected';
import { Providers, renderWithApolloData } from '@apollosproject/ui-test-utils';
import { MockedProvider } from '@apollo/client/testing';
import GET_LIKED_CONTENT from './getLikedContent';

import LikedContentFeedConnected from '.';

describe('LikedContentFeedConnected component', () => {
  it('renders LikedContentFeedConnected', async () => {
    const mock = {
      request: {
        query: GET_LIKED_CONTENT,
        variables: { first: 20 },
      },
      result: {
        data: {
          likedContent: {
            pageInfo: {
              __typename: 'PaginationInfo',
              endCursor: 'some-cursor',
            },
            __typename: 'ContentItemsConnection',
            edges: [
              {
                __typename: 'ContentItemsConnectionEdge',
                node: {
                  __typename: 'UniversalContentItem',
                  id: 'UniversalContentItem:4148aeb9482181025e9cad42826e676a',
                  hyphenatedTitle: 'Vacations and Your Budget',
                  title: 'Making Prayer a Spiritual Habit',
                  summary: 'Some summary',
                  coverImage: {
                    __typename: 'ImageMedia',
                    sources: [
                      {
                        __typename: 'ImageMediaSource',
                        uri:
                          'https://apollosrock.newspring.cc/GetImage.ashx?guid=55be1fd5-d8eb-43a6-aa70-f8e27bb63d31',
                      },
                    ],
                  },
                  isLiked: true,
                  sharing: {
                    title: 'Making Prayer a Spiritual Habit',
                    message: '',
                    url: 'https://apollosrock.newspring.cc/',
                  },
                  theme: null,
                },
              },
              {
                __typename: 'ContentItemsConnectionEdge',
                node: {
                  __typename: 'UniversalContentItem',
                  id: 'UniversalContentItem:9dbdc4e565dcaa3c9b17eb5ae42de96e',
                  hyphenatedTitle: 'Vacations and Your Budget',
                  title: 'Vacations and Your Budget',
                  summary: 'Some summary',
                  coverImage: {
                    __typename: 'ImageMedia',
                    sources: [
                      {
                        __typename: 'ImageMediaSource',
                        uri:
                          'https://apollosrock.newspring.cc/GetImage.ashx?guid=993efb47-47ec-4dec-a3d3-47bcacfbd58e',
                      },
                    ],
                  },
                  isLiked: true,
                  sharing: {
                    title: 'Vacations and Your Budget',
                    message: '',
                    url: 'https://apollosrock.newspring.cc/',
                  },
                  theme: null,
                },
              },
            ],
          },
        },
      },
    };

    const additionalMocks = mock.result.data.likedContent.edges.map(
      ({ node }) => ({
        request: {
          query: GET_CONTENT_CARD,
          variables: { contentId: node.id },
        },
        result: {
          data: {
            node: {
              ...node,
              coverImage: {
                name: 'Boom',
                ...node.coverImage,
              },
            },
          },
        },
      })
    );
    const navigation = { navigate: jest.fn() };
    const tree = await renderWithApolloData(
      <Providers
        MockedProvider={MockedProvider}
        mocks={[mock, ...additionalMocks]}
      >
        <LikedContentFeedConnected navigation={navigation} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
