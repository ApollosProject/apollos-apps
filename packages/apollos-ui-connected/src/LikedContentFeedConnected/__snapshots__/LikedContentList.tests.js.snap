// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LikedContentFeedConnected component renders LikedContentFeedConnected 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    avoidHeader={true}
    style={
      Object {
        "backgroundColor": "rgb(249, 249, 251)",
        "flex": 1,
        "height": "100%",
      }
    }
  >
    <RCTScrollView
      ListEmptyComponent={<ListEmptyComponent />}
      data={
        Array [
          Object {
            "__typename": "UniversalContentItem",
            "coverImage": Object {
              "__typename": "ImageMedia",
              "sources": Array [
                Object {
                  "__typename": "ImageMediaSource",
                  "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=55be1fd5-d8eb-43a6-aa70-f8e27bb63d31",
                },
              ],
            },
            "hyphenatedTitle": "Vacations and Your Budget",
            "id": "UniversalContentItem:4148aeb9482181025e9cad42826e676a",
            "isLiked": true,
            "summary": "Some summary",
            "theme": null,
            "title": "Making Prayer a Spiritual Habit",
          },
          Object {
            "__typename": "UniversalContentItem",
            "coverImage": Object {
              "__typename": "ImageMedia",
              "sources": Array [
                Object {
                  "__typename": "ImageMediaSource",
                  "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=993efb47-47ec-4dec-a3d3-47bcacfbd58e",
                },
              ],
            },
            "hyphenatedTitle": "Vacations and Your Budget",
            "id": "UniversalContentItem:9dbdc4e565dcaa3c9b17eb5ae42de96e",
            "isLiked": true,
            "summary": "Some summary",
            "theme": null,
            "title": "Vacations and Your Budget",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReached={[Function]}
      onEndReachedThreshold={0.7}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onPressItem={[Function]}
      onRefresh={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={false}
        />
      }
      refreshing={false}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <RCTRefreshControl />
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <View
              __typename="UniversalContentItem"
              contentId="UniversalContentItem:4148aeb9482181025e9cad42826e676a"
              coverImage={
                Object {
                  "__typename": "ImageMedia",
                  "sources": Array [
                    Object {
                      "__typename": "ImageMediaSource",
                      "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=55be1fd5-d8eb-43a6-aa70-f8e27bb63d31",
                    },
                  ],
                }
              }
              hyphenatedTitle="Vacations and Your Budget"
              id="UniversalContentItem:4148aeb9482181025e9cad42826e676a"
              isLiked={true}
              summary="Some summary"
              theme={null}
              title="Making Prayer a Spiritual Habit"
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <View
              __typename="UniversalContentItem"
              contentId="UniversalContentItem:9dbdc4e565dcaa3c9b17eb5ae42de96e"
              coverImage={
                Object {
                  "__typename": "ImageMedia",
                  "sources": Array [
                    Object {
                      "__typename": "ImageMediaSource",
                      "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=993efb47-47ec-4dec-a3d3-47bcacfbd58e",
                    },
                  ],
                }
              }
              hyphenatedTitle="Vacations and Your Budget"
              id="UniversalContentItem:9dbdc4e565dcaa3c9b17eb5ae42de96e"
              isLiked={true}
              summary="Some summary"
              theme={null}
              title="Vacations and Your Budget"
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
